staging/src/k8s.io/apiserver/pkg/endpoints/handlers/get.go

```diff
@@ -259,22 +260,32 @@ func ListResource(r rest.Lister, rw rest.Watcher, scope *RequestScope, forceWatc
 			}
 			klog.V(3).InfoS("Starting watch", "path", req.URL.Path, "resourceVersion", opts.ResourceVersion, "labels", opts.LabelSelector, "fields", opts.FieldSelector, "timeout", timeout)
 			ctx, cancel := context.WithTimeout(ctx, timeout)
-			defer cancel()
```

Comment:
is it possible for anything between here and the serve() invocation to panic? if so, we'll lose the cancel() call.

Reply:
That's possible, I add the defer back to call cancel if it returns early.

Reply:
I'm confused with this cancel, are we sure we need this complexity of returnEarly bool? context cancellation only propagates to childrens, no?

Reply:
+1 - I don't really follow what returnEarly gives us here...

Reply:
If either of `rw.Watch` or `serveWatchHandler`, will the ctx with cancellation leak since we don't defer calling cancel()?

Reply:
Followed @wojtek-t 's suggestion to make it easier to understand